inst/htmljs/animint.js

```diff
@@ -243,23 +245,38 @@ var animint = function (to_select, json_file) {
     // Save this geom and load it!
     update_geom(g_name, null);
   };
+
+var current_tr = plot_td.append("tr");
+
   var add_plot = function (p_name, p_info) {
-    // Each plot may have one or more legends. To make space for the
-    // legends, we put each plot in a table with one row and two
-    // columns: tdLeft and tdRight.
-    var plot_table = plot_td.append("table").style("display", "inline-block");
+    // console.log(p_info);
+    if(!current_tr) {
+    current_tr = plot_td.append("tr");
+  }
+  var td = current_tr.append("td");
+  var attributes = p_info.attributes || {};
+
+  td.attr("rowspan", attributes.rowspan || 1)
+    .attr("colspan", attributes.colspan || 1);
+    // Inner table for plot and legend
+    var plot_table = td.append("table").style("display", "inline-block");
     var plot_tr = plot_table.append("tr");
     var tdLeft = plot_tr.append("td");
-    var tdRight = plot_tr.append("td").attr("class", p_name+"_legend");
-    if(viz_id === null){
+    var tdRight = plot_tr.append("td").attr("class", p_name + "_legend");
+
+    // Set plot ID
+    if (viz_id === null) {
       p_info.plot_id = p_name;
-    }else{
+    } else {
```

Comment:
please click Files Changed tab, and undo any line changes which only involve white space, like this one